packages/firestore/src/core/firestore_client.ts

```diff
@@ -507,9 +508,31 @@ export function firestoreClientRunAggregationQuery(
   client: FirestoreClient,
   query: AggregateQuery
 ): Promise<AggregateQuerySnapshot> {
-  return client.asyncQueue.enqueue(() => {
-    return getAggregate(query);
+  const deferred = new Deferred<AggregateQuerySnapshot>();
+  client.asyncQueue.enqueueAndForget(async () => {
+    const remoteStore = await getRemoteStore(client);
+    if (!canUseNetwork(remoteStore)) {
+      logDebug(
+        LOG_TAG,
+        'The network is disabled. The task returned by ' +
```

Comment:
This debug message is somewhat misleading. First, the `getAggregateFromServerDirect()` function is going to be renamed. Second, it implies that the promise will be automatically resolved once network connectivity is restored. What do you think about just removing this log message altogether?